pkg/parser/enrich_date.go

```diff
@@ -29,40 +46,43 @@ func GenDateParse(date string) (string, time.Time) {
 	)
 
 	for _, dateFormat := range layouts {
-		t, err := time.Parse(dateFormat, date)
-		if err == nil && !t.IsZero() {
-			//if the year isn't set, set it to current date :)
-			if t.Year() == 0 {
-				t = t.AddDate(time.Now().UTC().Year(), 0, 0)
-			}
-			retstr, err := t.MarshalText()
-			if err != nil {
-				log.Warningf("Failed marshaling '%v'", t)
-				continue
-			}
-			return string(retstr), t
+		retstr, parsedDate := parseDateWithFormat(date, dateFormat)
+		if !parsedDate.IsZero() {
+			return retstr, parsedDate
 		}
 	}
-
-	now := time.Now().UTC()
-	retstr, err := now.MarshalText()
-	if err != nil {
-		log.Warning("Failed marshaling current time")
-		return "", time.Time{}
-	}
-	return string(retstr), now
+	return "", time.Time{}
 }
 
 func ParseDate(in string, p *types.Event, x interface{}) (map[string]string, error) {
 
 	var ret map[string]string = make(map[string]string)
-	tstr, tbin := GenDateParse(in)
-	if !tbin.IsZero() {
-		ret["MarshaledTime"] = tstr
+	var strDate string
+	var parsedDate time.Time
+
+	//log.Printf("received events : %+v", p.Parsed)
+	if v, ok := p.Parsed["DateFormat"]; ok && v != "" {
```

Comment:
I think it would be cleaner to have a `StrTimeFormat` at the root of the event object (and this will be consistent with `StrTime`).